PracticeTemplate/Controllers/Ex6/Ex6ViewController.swift

```diff
@@ -10,8 +10,50 @@ import UIKit
 
 class Ex6ViewController: BaseViewController {
     
+    // MARK: - IBoulet
+    @IBOutlet private weak var grayView: UIView!
+    @IBOutlet private weak var blueView: UIView!
+    @IBOutlet private weak var thumbView: UIView!
+    @IBOutlet private weak var thumbLabel: UILabel!
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        setupThumb()
+        setupLabel()
+    }
+    
+    // MARK: - Action
+    private func setupThumb() {
+        thumbView.layer.cornerRadius = thumbView.bounds.width / 2
+        thumbView.backgroundColor = .orange
+        thumbView.center = CGPoint(x: grayView.frame.midX, y: grayView.frame.midY)
+    }
+    private func setupLabel() {
+        thumbLabel.text = "%"
+        thumbLabel.textAlignment = .center
+    }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        if let touch = touches.first {
+            let location = touch.location(in: self.view)
+            if location.y >= grayView.frame.minY, location.y <= grayView.frame.maxY {
+                thumbView.center = CGPoint(x: grayView.center.x, y: location.y)
+                blueView.frame = CGRect(x: grayView.frame.origin.x, y: grayView.frame.origin.y, width: blueView.frame.width, height: location.y - grayView.frame.origin.y)
+                let percent = 100 * ( 1 - blueView.frame.height / grayView.frame.height)
```

Comment:
Số 1 này nghĩa là gì thế, 1 của cái gì.... space, padding hay pixel
```suggestion
                let magicNumber = 1
                let percent = 100 * (magicNumber - blueView.frame.height / grayView.frame.height)
```